ide/svg/src/org/netbeans/modules/svg/SVGPanel.java

```diff
@@ -0,0 +1,137 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.svg;
+
+import com.github.weisj.jsvg.SVGDocument;
+import com.github.weisj.jsvg.SVGRenderingHints;
+import java.awt.*;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.swing.*;
+
+/**
+ *
+ * @author Christian Lenz
+ */
+public class SVGPanel extends JPanel {
+
+    private static final Logger LOG = Logger.getLogger(SVGViewerElement.class.getName());
+
+    private SVGDocument svgDocument;
+    private double scale = 1.0D;
+    private BackgroundMode bgMode = BackgroundMode.DEFAULT;
+
+    public void setSVGDocument(SVGDocument doc) {
+        this.svgDocument = doc;
+
+        repaint();
+    }
+
+    public void setScale(double scale) {
+        this.scale = scale;
+
+        repaint();
+    }
+
+    public void setBackgroundMode(BackgroundMode mode) {
+        this.bgMode = mode;
+
+        repaint();
+    }
+
+    @Override
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+
+        Rectangle visibleRect = getVisibleRectangle();
+
+        switch (bgMode) {
+            case BLACK -> {
+                g.setColor(Color.BLACK);
+                g.fillRect(visibleRect.x, visibleRect.y, visibleRect.width, visibleRect.height);
+            }
+            case WHITE -> {
+                g.setColor(Color.WHITE);
+                g.fillRect(visibleRect.x, visibleRect.y, visibleRect.width, visibleRect.height);
+            }
+
+            case TRANSPARENT, DARK_TRANSPARENT ->
+                Utils.drawChestTilePattern(g, visibleRect, 20, bgMode == BackgroundMode.DARK_TRANSPARENT);
+            case DEFAULT -> {
+                g.setColor(getBackground());
+                g.fillRect(visibleRect.x, visibleRect.y, visibleRect.width, visibleRect.height);
+            }
+        }
+
+        if (svgDocument == null) {
+            return;
+        }
+
+        Graphics2D g2d = (Graphics2D) g.create();
+
+        try {
+            g2d.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+            g2d.setRenderingHint(RenderingHints.KEY_STROKE_CONTROL, RenderingHints.VALUE_STROKE_PURE);
+            g2d.setRenderingHint(SVGRenderingHints.KEY_MASK_CLIP_RENDERING, SVGRenderingHints.VALUE_MASK_CLIP_RENDERING_ACCURACY);
+
+            g2d.scale(scale, scale);
+
+            svgDocument.render(this, g2d);
+        } catch (Exception ex) {
+            LOG.log(Level.SEVERE, ex.getMessage());
```

Comment:
would be good if the panel would draw a text message to indicate that it can't render a svg.

I don't think we should log anything in an EDT draw method with level SEVERE since this is going to hit the notifications and will repeat. FINE or INFO is probably better here (FINE won't be logged by default - might be best? cc @matthiasblaesing ).